community/kernel/src/main/java/org/neo4j/kernel/impl/coreapi/TxStateTransactionDataSnapshot.java

```diff
@@ -219,10 +220,16 @@ private void takeSnapshot()
                             }
                         }
 
-                        node.get().labels().visitKeys( labelId ->
+                        node.get().labels().forEach( labelId ->
                         {
-                            removedLabels.add( new LabelEntryView( nodeId, store.labelGetName( labelId ) ) );
-                            return false;
+                            try
+                            {
+                                removedLabels.add( new LabelEntryView( nodeId, store.labelGetName( labelId ) ) );
+                            }
+                            catch ( LabelNotFoundKernelException e )
+                            {
+                                throw new IllegalStateException( "An entity that does not exist was modified.", e );
```

Comment:
please include node ids and label ids into exception message

Reply:
still this one

Reply:
fixed